cmd/kaniko-ecr/main.go

```diff
@@ -570,9 +569,12 @@ func createDockerConfig(dockerRegistry, dockerUsername, dockerPassword, accessKe
 			dockerConfig.SetCredHelper(ecrPublicDomain, "ecr-login")
 			dockerConfig.SetCredHelper(registry, "ecr-login")
 		}
+
+		// create config file from the credentials list
+		return dockerConfig.CreateDockerConfig(credentials, dockerConfigPath)
```

Comment:
If you do the create config here then it won't work. It should be outside of if else conditiom